lift/Readme.md

```diff
@@ -6,271 +6,360 @@
-least have a "forward" also known as `up` direction, which is the action of applying the migration. The reversal of a migration is commonly known as the `down`
-direction of the migration. While Prisma can automatically calculate the `down` for its built-in migrations, user-defined migration scripts need to define their
-own `down`.
+**TODO**
 
-### Declarative migration
+</details>
+<details>
+<summary>DeleteField</summary>
 
-An example for a declarative migration is this: You start with datamodel `A` and change it to datamodel `B`, while not providing the actual steps needed to
-achieve this change. The only thing you need to provide is the following:
+**TODO**
 
-`Datamodel A`
+</details>
+<details>
+<summary>UpdateField</summary>
 
-```graphql
-model User {
-  id: ID! @id
-  name: String
-}
-```
+**TODO**
 
-`Datamodel B`
+</details>
+<details>
+<summary>CreateEnum</summary>
 
-```graphql
-model User {
-  id: ID! @id
-  name: String!
-}
-```
+**TODO**
 
-In this example we made the `name` field required. The actual steps needed to achieve the change will be _inferred_ by Prisma. Therefore we can also talk about
-declarative migrations as **automatic migrations**.
+</details>
+<details>
+<summary>UpdateEnum</summary>
 
-### Imperative migration
+**TODO**
 
-In traditional migration systems as the one provided by [Active Record](https://edgeguides.rubyonrails.org/active_record_migrations.html), the actual change is
-instead expressed in an imperative manner:
+</details>
+<details>
+<summary>DeleteEnum</summary>
 
-```ruby
-class ChangeUserName < ActiveRecord::Migration[5.0]
-  def change
-    reversible do |dir|
-      change_table :User do |t|
-        t.string :name, null: false
-      end
-    end
-  end
-end
-```
+**TODO**
 
-With imperative migrations, the user needs to both calculate the "diff" in his head and write the needed actions down, which will result in the new schema.
+</details>
 
-While in Prismas model the declarative datamodel is the source of truth for migrations, Active Record takes the imperative migrations as the source of truth for
-migrations. Therefore one can call the imperative migrations also **manual migrations**.
+### Migration
 
-### Datamodel Snapshots
+A Migration is a grouping of one or more Steps. Migrations run in a transaction if the datasource allows it.
 
-As mentioned in [Declarative migration](#how-does-naming-of-migrations-work), the user adjusts the `datamodel.prisma` file, which results in a new migration
-folder when running `prisma migrate`. In order for the migration engine to know which changes need to be applied, it needs a "copy" of the datamodel. As this
-"copy" describes the datamodel in a specific point in time, we also refer to it as a "snapshot".
+### Hook
 
-### Migration Engine
+A hook is a custom shell script that runs either before or after the migration. Hooks give you full control over your migrations. You can write hooks to:
```

Comment:
It does not really give you full control imo. I would phrase it that it allows you to customise a migration when the normal migrations don't cut it.